test/framework/toy_build.py

```diff
@@ -271,6 +280,95 @@ def test_toy_broken(self):
         # cleanup
         shutil.rmtree(tmpdir)
 
+    def detect_log_file(self, tmp_log_path):
+        log_files = list(pathlib.Path(tmp_log_path).glob("**/*.log"))
+
+        self.assertTrue(len(log_files) >= 1, 'Log files generated')
+        log_file = log_files[0]
+        self.assertTrue(len(log_files) == 1, f"Log file '{log_file}' is unique")
+
+        return log_file
+
+    def assert_build_files_copied(self, buildpath, artifact_error_path):
+        build_dir = pathlib.Path(buildpath)
+        storage_dir = pathlib.Path(artifact_error_path)
+
+        app_build_dir = build_dir / r"toy/0.0/system-system"
+        iso_date_pattern = r"????????-??????"
+
+        for file in app_build_dir.glob("**/*"):
+            file_relative_path = file.relative_to(app_build_dir)
+            file_copies = list(storage_dir.glob(f"toy-0.0/{iso_date_pattern}/{file_relative_path}"))
+            self.assertTrue(len(file_copies) == 1, f"Unique copy of toy build file '{file}' made")
+            file_copy = file_copies[0]
+
+            if file_copy.is_file():
+                msg = f"File '{file}' copied successfully"
+                self.assertTrue(filecmp.cmp(str(file), str(file_copy), shallow=False), msg)
+
+    def assert_log_files_copied(self, log_file, log_error_path):
+        file_name = log_file.name
+        saved_log_files = list(pathlib.Path(log_error_path).glob(f"**/{file_name}"))
+        self.assertTrue(len(saved_log_files) == 1, f"Unique copy of log file '{log_file}' made")
+        for saved_log_file in saved_log_files:
+            msg = f"Log file '{log_file}' copied successfully"
+            self.assertTrue(filecmp.cmp(str(log_file), str(saved_log_file), shallow=False), msg)
+
+    def assert_error_reported(self, outtxt, output_regexs):
+        for regex_pattern in output_regexs:
+            regex = re.compile(regex_pattern, re.M)
+            msg_stdout = "Pattern %s found in full test report: %s" % (regex.pattern, outtxt)
+            self.assertTrue(regex.search(outtxt), msg_stdout)
```

Comment:
For 5.0x we can use:
```suggestion
            self.assertRegex(outtxt, regex)
```

Could also just use `regex_pattern` instead of `regex` if the flags are not required (here they are). This avoids the manually constructed message

Reply:
Done in the 5.0.x branch.